### ISSUE TITLE
aria-level, aria-posinset, and aria-setsize are not announced properties in MS Edge

### ISSUE THREAD

**Issue by maschad96** [nvaccess/nvda#10472]
### Steps to reproduce:
- Navigate to [Treeview Example B](https://www.w3.org/TR/wai-aria-practices/examples/treeview/treeview-2/treeview-2b.html)
- Launch Microsoft Edge or IE11
- Launch NVDA
- Attempt reading each link

### Actual behavior:
- The level of the current item, aria-setsize, and aria-posinset aren't being announced by NVDA

### Expected behavior:
- aria-level, setsize, and posinset should be used to announce the user's current position within the menu

#### NVDA installed/portable/running from source:
NVDA installed from nvaccess.org

#### NVDA version:
Version 2019.2.1

#### Windows version:
Windows 10 Enterprise, OS build 18362.418

#### Name and version of other software in use when reproducing the issue:
Microsoft Edge 44.18362.387.0
Internet Explorer 11

#### Other information about your system:
N/A

#### Does the issue still occur after restarting your PC?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No

**Comment by leonardder** [comment]
I'm afraid that, with Edge based on Chromium being on its way, this won't receive much attention for classic edge. Would you be able to use another browser to circumvent this?

**Comment by maschad96** [comment]
I understand your take. Even with the Edge 'preview' on the way, it would be best to cover the users who don't upgrade.

This issue applies to both IE11 and Edge, which put together is a good amount of users.

**Comment by leonardder** [comment]
Do you know how other screen readers like Narrator and JAWS treat this in IE and Edge?

**Comment by maschad96** [comment]
**Narrator**
- Narrator doesn't announce the level, position, or size with IE11.
- Narrator does announce the level, current position, and size of current level with Edge.

I'm working on getting a license for JAWS so that I can fill you in on that behavior.

**Comment by leonardder** [comment]
Or JAWS, you can use its 40 minutes trial.

**Comment by leonardder** [comment]
It looks like setsize and posinsetare properly exposed by Edge. This doesn't apply to the level.

As for setting position info according to these aria properties, that shouldn't be very difficult for Edge to be implemented. ARIA support in IE is fairly limited, so I wouldn't try to improve that.

**Comment by maschad96** [comment]
Regarding the discussion on JAWS with Edge, it appears to announce aria-level well when using declared properties. In chrome, it starts at 0 at the top-level even if explicitly declared as 1, continues to increment by 1 as it nests.

> As for setting position info according to these aria properties, that shouldn't be very difficult for Edge to be implemented. ARIA support in IE is fairly limited, so I wouldn't try to improve that.

JAWS announces aria-level, posinset, and setsize fairly well in IE, but I'm fine with keeping this issue solely revolving around Edge. We can always break off IE11 into a separate issue, but generally it would be great to see the improvement made to both.

Do you think with all the information we have, we can proceed investigating the solution for Edge?

**Comment by Adriani90** [comment]
Since Edge cromium will be distributed with Windows 10 20 H1 update, I guess we should concentrate on Edge Cromium.
Did you try the internet explorer simulation in Edge Cromium? Does it work there? Ideally using NVDA 2019.3.1.

**Comment by Adriani90** [comment]
Looking deeper to this example, in my view from an user perspective is not very best. It is ok to hear the position of every element, but the level should be reported only once when entering it. For example expanding "fruits" with right arrow, it says "expanded". Pressing down arrow, it says "level 2". After that it is self explanatory that all elements thereunder are level 2, unless I press down arrow multiple times until the level is changed. So the level should not be reported on every element of the tree.
But this is only my subjective opinion.

**Comment by seanbudd** [comment]
Closing in favor of the newer, and more generic #14411

**Comment by jaxcze** [comment]
Hi,

Try my add-on: **Last Item Announcer**. It's a relatively complex add-on that addresses this issue: [https://nvda-addons.janbalak.name/last-item-announcer/en](https://nvda-addons.janbalak.name/last-item-announcer/en)

It is not yet available in the Add-on Store, but it is available on my website.

Best regards,
Jan Balák
